pylops/signalprocessing/_nonstatconvolve2d_cuda.py

```diff
@@ -3,7 +3,7 @@
 from numba import cuda
 
 
-@cuda.jit
+@cuda.jit(max_registers=40)
```

Comment:
Fixes https://stackoverflow.com/questions/68658970/why-launching-a-numba-cuda-kernel-works-with-up-to-640-threads-but-fails-with-6

Reply:
I must say I didn't face this issue but:

- I used a pretty good GPU for all of my testing (RTX 3090).
- I didn't really make any model bigger than then one at the bottom of this notebook https://github.com/PyLops/pylops_notebooks/blob/master/developement-cupy/LeastSquaresMigration_imagedomainmarmousi.ipynb

Having said that, lets keep this in and re-evaluate if it is possible to make it more dynamic by finding out properties of a GPU later on

Reply:
I had to dig really deep to find this fix. I am not sure what causes it. Just to leave it registered, removing the [final lines](https://github.com/cako/pylops/blob/ce8a0151cfb6b36a12b5a1f5f6f9e3a5d862bf4c/pylops/signalprocessing/_nonstatconvolve2d_cuda.py#L79-L82) makes the code run. I tried to manually remove the number of registers but it did not work. The only fix I found was the one above. We should definitely revisit this as I don't fully understand how max_registers imposes this limit.